File: Week4/.NetHw/PageObject/HomePage.cs
using System;
using OpenQA.Selenium;
using NUnit.Framework;
using SeleniumExtras.PageObjects;

namespace _NetHw {
public class HomePage  : BasePage {

public HomePage (IWebDriver driver) {
      HomePage.driver = driver;
      PageFactory.InitElements(driver,this);
  }

   public String login() {
    driver.FindElement(By.CssSelector(".user-panel-icon-text")).Click();
    driver.FindElement(By.Name("email")).SendKeys("testhesabi211@gmail.com");
    driver.FindElement(By.Name("password")).SendKeys("Deneme.123");
    driver.FindElement(By.Id("pw-search-input")).SendKeys("ayakkabı");
    driver.FindElement(By.Id("pw-search-input")).SendKeys(Keys.Enter);
    string assertText2 = driver.FindElement(By.XPath("(//h1[@class='fsize-36 font-financier'])[1]")).Text;
    if (assertText2 != "\\\"ayakkabı\\\" için sonuçlar") {
      Assert.Fail();
    }
    return assertText2;
  }
}
}

File: Week4/pythonProject/Pages/HomePage.py
from selenium.webdriver.common.by import By
from selenium.webdriver.common.keys import Keys



class HomePage():
    def __init__(self, driver):
        self.driver = driver
        self.searchBox_css = "*[data-test=\"search-input\"]"


    def login(self, param, param1):
        self.driver.get("https://www.dsdamat.com/")
        self.driver.find_element(By.LINK_TEXT, "Üye Girişi").click()
        self.driver.find_element(By.NAME, "email").send_keys(param)
        self.driver.find_element(By.NAME, "password").send_keys(param1)
        self.driver.find_element(By.ID, "pw-search-input").send_keys("ayakkabı")
        self.driver.find_element(By.ID, "pw-search-input").send_keys(Keys.ENTER)
        self.driver.close()
        pass
